src/qibolab/instruments/abstract.py

```diff
@@ -46,6 +46,23 @@ def stop(self):
     def disconnect(self):
         raise NotImplementedError
 
+    def play(self, *args, **kwargs):
+        """Play a pulse sequence and retrieve feedback."""
+        raise NotImplementedError(f"Instrument {self.name} does not support play.")
+
+    def sweep(self, *args, **kwargs):
+        """Play a pulse sequence while sweeping one or more parameters."""
+        raise NotImplementedError(f"Instrument {self.name} does not support sweep.")
+
+
+class LocalOscillator(AbstractInstrument):
+    """Abstraction for local oscillator instruments.
+
+    Local oscillators are used to upconvert signals, when
+    the controllers cannot send sufficiently high frequencies
+    to address the qubits and resonators.
+    """
+
```

Comment:
I believe this class or even better the `AbstractInstrument` is missing some try/except mechanism for connect/start/stop/setup.

Reply:
This is already done in some of the instruments, such as the Rohde-Schwarz

https://github.com/qiboteam/qibolab/blob/0f19c4f5812f5b24c394e917a8c16f321193a416/src/qibolab/instruments/rohde_schwarz.py#L51

We could do it at the `AbstractInstrument` level so that it is inherited everywhere, but I am not sure what is a clean way to do this. One possibility is to have an additional `def _connect:` that the subclasses override or even define a decorator that implements try-except.

Also, I am not sure how useful it will be to have this at the `AbstractInstrument` because connection works differently for each instrument. For example I noticed that if the QM instruments are not available, the call to connect to them just hangs, so even if we use try-except we will never fall to the except.